xbmc/platform/android/activity/XBMCApp.cpp

```diff
@@ -1250,6 +1250,18 @@ int CXBMCApp::WaitForActivityResult(const CJNIIntent &intent, int requestCode, C
     result = event->GetResultData();
     ret = event->GetResultCode();
   }
+
+  // delete from m_activityResultEvents map before deleting the event
+  CSingleLock lock(m_activityResultMutex);
+  for (auto it = m_activityResultEvents.begin(); it != m_activityResultEvents.end(); ++it)
+  {
+    if ((*it)->GetRequestCode() == requestCode)
+    {
+      m_activityResultEvents.erase(it);
+      break;
+    }
+  }
```

Comment:
know the v20 pr looks the same and there is no technical difference, but shouldn't we do it like this usually?
```c
for (const auto& activityResultEvent : m_activityResultEvents)
{
  if (activityResultEvent->GetRequestCode() == requestCode)
  {
    m_activityResultEvents.erase(activityResultEvent);
    break;
  }
}
```

Reply:
Yes C+11 iterators please

Reply:
+1

Reply:
Thanks will do